### PR TITLE
DATAUP-710 add error checking for missing/extra bulk import spreadsheet columns

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

As it says, if a user uploads a spreadsheet that has extra parameter ids or missing parameter ids, we need to throw errors for that and prevent opening an app cell. Theoretically, if a user works from a template, this will never happen, but we should check anyway.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-X
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook
- [x] Any dependent changes have been merged and published in downstream modules

# Updating Version and Release Notes (if applicable)

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: kbase-extension/static/kbase/js/widgets/narrative_core/upload/importSetup.js [hunk 1]
```diff
@@ -219,9 +227,9 @@ define([
          *   }
          * }
          */
-        const typeToSpecParams = Object.entries(appIdToType).reduce(
-            (typeToSpecParams, [appId, dataType]) => {
-                const spec = appIdToSpec[appId];
+        const typeToAlteredParams = Object.entries(appIdToType).reduce(
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/narrative_core/upload/importSetup.js:230]: sonar cloud says that this is already defined in the upper scope

File: test/unit/spec/narrative_core/upload/importSetup-spec.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ define([
     const stagingServiceUrl = Config.url('staging_api_url');
     const RELEASE_TAG = 'release';
 
-    describe('ImportSetup module tests', () => {
+    fdescribe('ImportSetup module tests', () => {
```
Comment (ialarmedalien) [inline test/unit/spec/narrative_core/upload/importSetup-spec.js:17]: `rm f`

File: test/unit/spec/narrative_core/upload/importSetup-spec.js [hunk 3]
```diff
@@ -286,138 +286,268 @@ define([
...
+                        await ImportSetup.setupImportCells(importInputs);
+                    } catch (e) {
+                        error = e;
+                    }
+                    // make sure we get the right list, and that the fileErrors
+                    // look like : { file1: [error], file2: [error] }
+                    expect(error).toEqual(jasmine.any(Errors.ImportSetupError));
+                    [file1, file2].forEach((fileName) => {
+                        expect(error.fileErrors[fileName]).toEqual(jasmine.any(Array));
+                    });
```
Comment (ialarmedalien) [inline test/unit/spec/narrative_core/upload/importSetup-spec.js:422]: is it possible to test the error structure more completely? Do you know what is going to be in `error.fileErrors`?

Comment (briehl) [reply test/unit/spec/narrative_core/upload/importSetup-spec.js:422]: Ah, I thought I committed that change. Ugh, will do so in a few minutes.

File: kbase-extension/static/kbase/js/widgets/narrative_core/upload/importSetup.js (hunk 1 above)
Comment (briehl) [reply kbase-extension/static/kbase/js/widgets/narrative_core/upload/importSetup.js:230]: Oh, I see how I named things. Fixed!

File: test/unit/spec/narrative_core/upload/importSetup-spec.js (hunk 2 above)
Comment (briehl) [reply test/unit/spec/narrative_core/upload/importSetup-spec.js]: Jeez, someone sure is extra sloppy today!

**Review by ialarmedalien — APPROVED** [review]
The test failures seem are unrelated to this code, so LGTM.

**Pull request merged**